src/validation.h

```diff
@@ -430,21 +430,21 @@ class CVerifyDB {
 /** Replay blocks that aren't fully applied to the database. */
 bool ReplayBlocks(const CChainParams& params, CCoinsView* view);
 
-inline CBlockIndex* LookupBlockIndex(const uint256& hash)
+inline CBlockIndex* LookupBlockIndex(const uint256& hash) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
```

Comment:
Can we just take cs_main inside instead of requiring it? Seems like that would massively reduce the diff size here.

Reply:
Oh, very good point! Now fixed!

Reply:
Don't we want to avoid recursive locks?